### PR TITLE
Updated NPM lockfile

### LINKED ISSUE #468: Confirm compatibility with WordPress 6.6
## Summary

Confirm compatibility with WordPress 6.6.

### Team
Primary Assignee: @mchirag2002 
PR reviewer: @SH4LIN 
QA: @maharrshi 

## References

- https://make.wordpress.org/core/6-6/

## Acceptance Criteria

- As a WordPress engineer, I can use Elementary to scaffold a brand new project with WordPress 6.6
- New features work as expected
- Deprecations are gracefully handled

```[tasklist]
## Additional Tasks
- [ ] Create testing methodology documentation for WP upgrade testing (which can be used going forward) @mchirag2002
- [ ] Support for Interactivity API @SH4LIN
- [ ] Run PHPCS scan @mchirag2002
- [ ] Make sure that README is up to date @mchirag2002
  - [ ] @SH4LIN Add documentation re: how to approach developing a new block
- [ ] Confirm maximum PHP compatibility @SH4LIN
- [ ] Raise PR to address issues encountered https://github.com/rtCamp/theme-elementary/issues/468#issuecomment-2235807396 @mchirag2002
- [ ] Raise PR(s) to address issues encountered https://github.com/rtCamp/theme-elementary/issues/468#issuecomment-2235807396 @mchirag2002
```

### PR DESCRIPTION
**Pull request by mchirag2002**
## Description

This task involves updating node packages without breaking any existing functionalities.

## Technical Details

- Updated `package.json` to include the latest dependencies without breaking the existing functionality.
- Added @formatjs/icu-messageformat-parser as a new dependency

## Fixes/Covers issue

Fixes #468

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,5 +61,8 @@
 		"pretest:php": "wp-env run cli --env-cwd=/var/www/html/wp-content/themes/$(basename $(pwd)) composer install --no-interaction --no-scripts",
 		"test:php": "wp-env run cli --env-cwd=/var/www/html/wp-content/themes/$(basename $(pwd)) composer exec 'phpunit --do-not-cache-result --verbose'",
 		"wp-env": "wp-env"
+	},
+	"dependencies": {
+		"@formatjs/icu-messageformat-parser": "2.7.8"
```
Comment (RahiDroid) [inline package.json:66]: Should this be part of `devDependencies`?

Comment (thelovekesh) [reply package.json:66]: Why this additional dependency is required?

Comment (mchirag2002) [reply package.json:66]: @thelovekesh the `intl-messageformat-parser` which was being used has been deprecated now and this `@formatjs/icu-messageformat-parser` was suggested as the new updated version of it. Hence the update.

image

Comment (thelovekesh) [reply package.json]: @mchirag2002 Such npm warnings are common. I don't see a reason why we need to include another dependency to just resolve this warning even we don't use it in our project directly.

Comment (RahiDroid) [reply package.json]: @mchirag2002 

@thelovekesh is right, if it's another package's dependency, that package should update it. I didn't realize it's a dependency not being used directly by us.

Comment (thelovekesh) [reply package.json]: Even with this change, I am getting these warnings:

image

and this package is being used by

```
➜  theme-elementary git:(feature/update-node-modules) npm ls intl-messageformat-parser          
elementary-theme@1.0.0 /home/thelovekesh/Desktop/work/rtCamp/theme-elementary
└─┬ @wordpress/scripts@28.3.0
  └─┬ @wordpress/e2e-test-utils-playwright@1.3.0
    └─┬ lighthouse@10.4.0
      └─┬ intl-messageformat@4.4.0
        └── intl-messageformat-parser@1.8.1
```

**Review by thelovekesh — CHANGES_REQUESTED** [review]
Thanks @mchirag2002! Please take a look at https://github.com/rtCamp/theme-elementary/pull/479#discussion_r1682761256 and https://github.com/rtCamp/theme-elementary/pull/479#discussion_r1682766485.

File: package.json (hunk 1 above)
Comment (mchirag2002) [reply package.json]: @thelovekesh Implemented the requested changes in - https://github.com/rtCamp/theme-elementary/pull/479/commits/e068701df05026a1e3efe6a452219aac3192d887

**Review by thelovekesh — APPROVED** [review]

**Pull request merged**